contracts/HATVaultsTokenWrapper.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/utils/SafeERC20.sol";
+
+
+ // Wrap an ERC20 token with other ERC20 token.
+contract HATVaultsTokenWrapper {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable wrappedToken;
+    address public immutable poolContract;
+     constructor(address _tokenToWrap, address _poolContract) {
+        wrappedToken = IERC20(_tokenToWrap);
+        poolContract = _poolContract;
+
+     }
+
+  function transferFrom(address _sender, address _recipient, uint256 _amount) public returns (bool) {
+    require(msg.sender == poolContract, "only poolContract can call transferFrom");
```

Comment:
also uniswap v3 should be able to transferFrom https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L773 . this is the reason it is approve at https://github.com/hats-finance/hats-contracts/blob/develop/contracts/HATVaults.sol#L755